### PR TITLE
Update icon in home button

### LINKED ISSUE #65428: Site editor: iterate on site icon animation
[While this initial PR](https://github.com/WordPress/gutenberg/pull/63986#) landed to try to clarify that the site icon is a back button, I think it's missing the mark and looks more broken than anything else. @WordPress/gutenberg-design can we iterate here? I think we can do better for 6.7. At the least, I think we should revert to what was there before. If we can, we should iterate.

https://github.com/user-attachments/assets/7d76eba6-bacb-47ea-baf1-957cecc25a52

### PR DESCRIPTION
**Pull request by jameskoster**
Closes #65428.

Screenshot 2024-09-19 at 15 54 49

## What?
Replace the current icon in the home button.

## Why?
See #65428.

## How?
In this PR I swapped to the `menu` icon. It's not perfect because the explicit metaphor may not fully capture how this element might evolve in the future (absorbing admin bar functionality). However, it is a very familiar icon, and aligns with the current tooltip 'Open navigation'.

Other options I ruled out:

* `<` isn't appropriate because it suggests going _back_, which isn't always the resulting behavior. We've already iterated away from this once (https://github.com/WordPress/gutenberg/pull/64161).
* The house icon has additional meaning that might not be useful in this context. Users might confuse it as a link to view their homepage.

We can explore new alternative icons here too.

One option could be something like the 'exit full screen' icon in macOS (green button):

Screenshot 2024-09-19 at 10 31 16

Another option could be a diagonal arrow:

Screenshot 2024-09-19 at 16 46 29
Screenshot 2024-09-19 at 16 47 22

### CODE REVIEW

**Comment by jameskoster** [comment]
I pushed the diagonal arrow approach to the PR, give it a try :) 

Screenshot 2024-09-19 at 17 33 22

**Comment by richtabor** [comment]
This doesn't seem right pointing down to the content. Perhaps top left (NW). 

CleanShot 2024-09-19 at 14 30 35

**Comment by afercia** [comment]
I'm not sure a diagonal arrow helps understand the functionality.
A few more issues:
- It only works on hover. Nothing happens on focus.
- There are two tab stops when navigating with the keyboard. In fact, the `edit-site-editor__view-mode-toggle` container has a `tabindex=0` attribute, not sure why. Although this comes from previous changes it must be fixed for the release as a silent tab stop is not OK.
- The focus style is insufficient, it's barely visible.
- The tooltip position is inconsistent across Post Editor and Site editor, not sure why (see screenshot).
- 
site editor

**Comment by jameskoster** [comment]
@afercia agree those are issues to fix, but probably not here.

More options:

Screenshot 2024-09-20 at 16 57 08
Screenshot 2024-09-20 at 16 57 58

**Comment by stokesman** [comment]
> In this PR I swapped to the `menu` icon. It's not perfect because the explicit metaphor may not fully capture how this element might evolve in the future (absorbing admin bar functionality)

The "menu" icon seems as good as it gets for now. Regarding future concerns, if admin bar functionality would stretch the metaphor too much perhaps such functions merit their own UI.

**Comment by jasmussen** [comment]
The diagonal arrow works well, that's a good start, and we can go from there.

**Comment by afercia** [comment]
> The diagonal arrow works well,

I kindly disagree. This button opens a manu. The icon should represent the concept of a menu. The most common pattern I can think of for such an icon is the so called 'hamburger' icon. As a user, a diagonal arrow doesn't mean much to me. I'm not sure why we'd want to go fancy with obscure, meaningless icons instead of choosing one that clearly represents the concept of a menu.

**Comment by jameskoster** [comment]
Pushed the alternative arrow to try. 

Based on the current behavior I think the `menu` icon works better, but I'm happy to disagree and commit here. The diagonal arrow feels like an improvement on trunk which is what matters most.

**Comment by afercia** [comment]
> Based on the current behavior I think the `menu` icon works better, but I'm happy to disagree and commit here. The diagonal arrow feels like an improvement

Seems to me the prevalent opinion on this change is to use the `menu` icon. In a collaborative open source project the prevalent opinion of contributors is what determines implementation choices.

**Comment by jasmussen** [comment]
The problem with the menu item is that it can easily be confused with the nav menu block, and even navigation elements for the site itself.

**Review by mtias — APPROVED** [review]
There's a visual tension as well between the list view icon and the hamburger in such close proximity. A hamburger menu generally opens in place while the post editor doesn't do that just yet, triggering a full navigation instead, so I think the arrow is a bit more adequate in the current context. We can revisit this during beta if necessary.

File: packages/icons/src/library/direction-nw.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * WordPress dependencies
+ */
+import { SVG, Path } from '@wordpress/primitives';
+
+const directionNW = (
```
Comment (mtias) [inline packages/icons/src/library/direction-nw.js]: I think these should be named `arrowTopLeft` and such for consistency with existing icons.

**Comment by stokesman** [comment]
> There's a visual tension as well between the list view icon and the hamburger in such close proximity

I totally agree. Though how does one qualify how much it frustrates actual use?

> A hamburger menu generally opens in place while the post editor doesn't do that just yet, triggering a full navigation instead

True, yet this seems unlikely to frustrate usage. It is still a good point for broader consideration and it sounds as if we are planning to not have it be a full navigation at some point.

The arrow is better than the current box so that’s a win. Yet not using the menu icon still feels like a miss of the practical mark because we’re aiming at perfection (which is currently out of range).

**Comment by mtias** [comment]
@stokesman yes, for sure, let's not consider this final

**Comment by afercia** [comment]
> There's a visual tension as well between the list view icon and the hamburger in such close proximity.

Yes I'd agree. However, the List View isn't just a list view any longer. The panel contains the List View _and_ the Outline. As such, the toggle button that opens a panel with List View + Outline should be renamed and the icon changed.
That should be considered together with a follow-up iteration for the home button.

**Pull request merged**